Added app.ts

fsarres opened the pull request.

fsarres committed:
added app.ts

fsarres committed:
Added index.ts

Ordervous reviewed (changes requested):
Felipe, 

tsconfig - your version of javascript is very old, we should not update to an older version. leave es2021. 
I think in the course we were using even es6. Not sure how you've got es5. 

Index.ts - this is the wrong kind of code for the model index. I can show you what we need for this file. 

Felipe.js - we might delete these test files. 

app.ts - not sure about using mysql12 here, we didn't really do that on course towards the end and having the database connection here instead of in the models/index. 
We should make the DB name something else too like dadsDB. 
From where you start with:

const app = express()  

Above it none of this is in the app.ts. 
Below it and most of it should be but its different than what we used a little bit? 

This is what I have for the app.ts file:

import morgan from "morgan";
import express, { NextFunction, Request, Response } from 'express'
import { db } from './models';
import tweetRoutes from './routes/tweetRoutes'
import userRoutes from './routes/userRoutes'


const app = express();

app.use(morgan('dev'));

app.use(express.json());
app.use(express.urlencoded({extended: true}));

const cors = require('cors')
app.use(cors())

// routes
app.use('/api/tweets', tweetRoutes);
app.use('/api/users', userRoutes);

app.use((req: Request, res: Response, next: NextFunction) => {
    res.status(405).end();
});

// Syncing our database
db.sync().then(() => {
    console.info("connected to the database!")
});

app.listen(3000);


And models/index.ts :

import { Sequelize } from "sequelize";
import { UserFactory } from "./user";
import { TweetFactory } from "./tweet";
import { AssociateUserTweets } from "./tweet";

const dbName = 'twitterDB';
const username = 'root';
const password = 'Password1!';

const sequelize = new Sequelize(dbName, username, password, {
    host: 'localhost',
    port: 3306,
    dialect: 'mysql'
});

TweetFactory(sequelize);
UserFactory(sequelize);
AssociateUserTweets();

export const db = sequelize;

fsarres committed:
deleted git test files

fsarres commented:
I need to know the name of the rest of the components/functions so I added just generic code to replace it later. The index.ts had dummy code which would be changed later, the app.ts I made some changes but I need the other parts to complete it. I manually changed es2021 to es5 because that's the version the lesson had, es5, but I changed it back to es2021. I tried with mysql2 because vscode asked to upgrade and I found online to try mysql2 instead, but I changed it back to mysql, not sure why I got that error.

fsarres committed:
added changes to index.ts and app.ts

fsarres commented:
I did not add the components such as routes/tweetRoutes because I don't know the name of the components/functions yet but I added them as you suggested, I was thinking about modifying it later. The index.ts had sample code which I'd modify later, the app.ts was pretty much similar (I took as guidance the example from the lesson) but I added the minor changes.

I manually changed the es2021 to es5 because that's what the lesson had but I changed it back and also modified the database name.

fsarres committed:
added index.ts
